File: 101 Symmetric Tree/rec.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    # @param {TreeNode} root
    # @return {boolean}
    def isSymmetric(self, root):
        if root==None:
            return True
        return self.rechelper(root.left,root.right)
    def rechelper(self,root1,root2):
        if root1==None and root2==None:
            return True
        if root1==None or root2==None:
            return False
        if root1.val!=root2.val:
            return False
        return self.rechelper(root1.left,root2.right) and self.rechelper(root1.right,root2.left)

File: 101 Symmetric Tree/rec.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    bool isSymmetric(TreeNode* root) {
        if (root==NULL)
            return true;
        return rechelper(root->left,root->right);
    }
    bool rechelper(TreeNode* left,TreeNode* right){
        if (left==NULL&&right==NULL)
            return true;
        if (left==NULL||right==NULL)
            return false;
        if (left->val!=right->val)
            return false;
        return rechelper(left->left,right->right)&&rechelper(left->right,right->left);
    }
};